src/modules/core/sagas/messages.ts

```diff
@@ -0,0 +1,55 @@
+import nanoid from 'nanoid';
+import { call, put, take } from 'redux-saga/effects';
+
+import { putError } from '~utils/saga/effects';
+import { ActionTypes } from '~redux/index';
+import { AllActions } from '~redux/types/actions';
+import { Context, getContext } from '~context/index';
+
+export function* signMessage(purpose, message) {
+  const wallet = yield getContext(Context.WALLET);
+  const messageId = `${nanoid(10)}-signMessage`;
+  /*
+   * @NOTE Initiate the message signing process
+   */
+  const messageString = JSON.stringify(message);
+  yield put<AllActions>({
+    type: ActionTypes.MESSAGE_CREATED,
+    payload: {
+      id: messageId,
+      purpose,
+      message: messageString,
+      createdAt: new Date(),
+    },
+  });
+
+  /*
+   * @NOTE Wait (block) until there's a matching action
+   * and get its generated id for the async listener
+   */
+  const {
+    meta: { id },
+  } = yield take(
+    (action: AllActions) =>
+      action.type === ActionTypes.MESSAGE_SIGN &&
+      action.payload.id === messageId,
+  );
+  try {
+    const signature = yield call([wallet, wallet.signMessage], {
+      message: messageString,
+    });
+
+    yield put<AllActions>({
+      type: ActionTypes.MESSAGE_SIGNED,
+      payload: { id: messageId, signature },
+      meta: { id },
+    });
+    return signature;
+  } catch (caughtError) {
```

Comment:
This works fine for Metamask and Hardware wallets, which can cancel signing a transaction _"externally"_:

![Screenshot from 2019-08-30 10-07-50](https://user-images.githubusercontent.com/1193222/64000408-58b8d680-cb0e-11e9-8d0d-2115e7f84746.png)

But won't work for software wallet, which rely on our Gas Station interface to cancel said message signing

![Screenshot from 2019-08-30 10-08-18](https://user-images.githubusercontent.com/1193222/64000447-7128f100-cb0e-11e9-8854-f4a727fb094e.png)
![Screenshot from 2019-08-30 10-08-26](https://user-images.githubusercontent.com/1193222/64000449-71c18780-cb0e-11e9-845d-1ee077bb7144.png)

I think this is a case where we need to `throw` when cancelling a message signing from the gas station

Reply:
Nice catch. I've fixed it and will push soon